zimui/src/components/DevdocsNavbar.vue

```diff
@@ -0,0 +1,229 @@
+  isSelected: boolean
+}
+
+interface SectionEntry {
+  name: string
+  isOpen: boolean
+  children: PageEntry[]
+}
+
+interface Document {
+  name: string
+  landingHref: string
+  isSelected: boolean
+  licenseHref: string
+  version: string
+  children: SectionEntry[]
+}
+
+// Is there an error rendering?
+const error = ref('')
+// Navigation to display.
+const navigation = ref<Document>({
+  name: 'Loading...',
+  landingHref: '#',
+  isSelected: false,
+  licenseHref: '',
+  version: '',
+  children: []
+})
+// Currently selected page which may differ from the original if the user clicks a link
+// in the sidebar.
+const currentPage = ref(props.current)
+// Map of section name to open state if opened/closed by the user.
+const sectionOpenStates = ref(new Map<string, boolean>())
+
+// Make a link relative using props.prefix.
+function makeRelative(link: string): string {
+  return props.prefix + link
+}
+
+// Set of all links in the navbar that can be highlighted.
+const allHrefs = computed(() => {
+  const hrefList = navigation.value.children
+    .map((section) => section.children)
+    .flat()
+    .map((page) => page.href)
+
+  // Push in the root href (should always be "index")
+  hrefList.push(navigation.value.landingHref)
+
+  return new Set(hrefList)
+})
+
+// Get the href that most closely represents the current page so it can be highlighted.
+const highlightedHref = computed(() => {
+  const currentPageWithHash = currentPage.value + window.location.hash
+
+  if (allHrefs.value.has(currentPageWithHash)) {
+    return currentPageWithHash
+  }
+
+  return currentPage.value
+})
+
+// Navigation state to be rendered.
+const displayNavigation = computed(() => {
+  const output: Document = {
+    name: navigation.value.name,
+    version: navigation.value.version,
+    landingHref: makeRelative(navigation.value.landingHref),
+    licenseHref: makeRelative(navigation.value.licenseHref),
+    isSelected: navigation.value.landingHref === highlightedHref.value,
+    children: []
+  }
+
+  for (var section of navigation.value.children) {
+    const sectionEntry: SectionEntry = {
+      name: section.name,
+      isOpen: false,
+      children: []
+    }
+
+    var hasSelectedChild = false;
+    for (var page of section.children) {
+      const pageEntry: PageEntry = {
+        name: page.name,
+        href: makeRelative(page.href),
+        isSelected: page.href === highlightedHref.value
+      }
+
+      hasSelectedChild = hasSelectedChild || pageEntry.isSelected;
+
+      sectionEntry.children.push(pageEntry)
+    }
+
+    // Sections take on the last state a person set on them.
+    // By default, they're open if one of their children is selected.
+    sectionEntry.isOpen = sectionOpenStates.value.get(sectionEntry.name) ?? hasSelectedChild
+
+    output.children.push(sectionEntry)
+  }
+
+  return output
+})
+
+// Load the navigation data when the component is rendered.
+onMounted(() => {
+  // Kiwix doesn't like fetch so use XHR instead.
```

Comment:
This is working normally with other scrapers using Vue.JS, with axios library (which I'm pretty sure is not issuing XHR, but I might be wrong). Why not using axios?

Reply:
It looks like axios is using XHR under the hood according to their main page: https://axios-http.com/docs/intro. I first tried with fetch because it's more modern but got a bunch of errors on stdout when using it in Kiwix. 

I can swap this to axios to be more idiomatic.